classes/items/itembase.hxx

```diff
@@ -9,30 +9,46 @@
 #define ITP298_CAPSTONE_ITEMBASE_HXX
 
 #include <string>
-
+#include <sstream>
 
 enum item_types {WEAPON = 0, ARMOR}; //allows the program to decide what type of item is being generated
+enum item_tiers {COMMON = 0, UNCOMMON, RARE, EPIC, LEGENDARY};
 
 class Item {
 
 //Private variables
 private:
-    std::string name; //used to denote the name of an item
-    unsigned type; //used to denote the type of item upon creation
-    unsigned value; // used to denote the value of a generated item
+    std::string name; //used to denote the name of an item.
+    unsigned int type; //used to denote the type of item upon creation.
+    unsigned int value; // used to denote the value of a generated item.
+    unsigned int tier; // used to denote the tier (rarity) of an item.
+    unsigned int level;  //used to determine the level of an item.
 
     //private functions
     void generate(); //item generation function
 
 
 public:
-    Item(); //calls constructor
+    Item(std::string name,
+         unsigned int type,
+         unsigned int tier,
+         unsigned int value,
+         unsigned int level);
+
+    this->name = name; //assigns the given value of name
+    this->type = type;  //assigns the given value of type
+    this->value = value; //assigns the given value of value
+    this->tier = tier; //assigns the given value of tier
+    this->level = level; //assigns the given value of level
```

Comment:
Copy this to the .cpp file. And switch the ; after "level" with a {. And then put a } at the end of the body. And you should be good to go with compiling.